proxy-identity/main.go

```diff
@@ -23,19 +22,17 @@ const (
 )
 
 func main() {
-	cmd := flag.NewFlagSet("public-api", flag.ExitOnError)
+	defer runProxy()
 
-	name := cmd.String("name", "", "identity name")
-	dir := cmd.String("dir", "", "directory under which credentials are written")
-
-	flags.ConfigureAndParse(cmd, os.Args[1:])
+	name := os.Getenv("LINKERD2_PROXY_IDENTITY_LOCAL_NAME")
+	dir := os.Getenv("LINKERD2_PROXY_IDENTITY_DIR")
```

Comment:
nit: could we move these below the check for `envDisabled`? If identity is disabled we won't need these variables and they most likely will not even be set.